xdcs-server/src/main/java/pl/edu/agh/xdcs/restapi/mapper/TaskMapper.java

```diff
@@ -35,9 +42,15 @@ public TaskDto toRestEntity(Task model) {
         TaskDto dto = new TaskDto();
         dto.setId(model.getId());
         dto.setName(model.getName());
-        dto.setState(stateMapper.toApiEntity(model.getType()));
+        dto.setState(getTaskState(model));
         dto.setDeploymentId(model.getDeploymentDescriptor().getDeploymentRef().getReferencedObjectId());
         dto.setTimeCreated(model.getTimeCreated().atOffset(userContext.getCurrentZoneOffset()));
         return dto;
     }
+
+    private TaskDto.StateEnum getTaskState(Task model) {
```

Comment:
tutaj masz `stateMapper` więc w sumie to on powinien mapować stan i brać pod uwagę czy result jest ustawiony czy nie